src/CoamClient.js

```diff
@@ -436,6 +436,21 @@ class CoamClient {
     }
 
 
+    getUsersWithResource(resourceType, permissionFilters) {
+        let url = this.__buildUrl(`/auth/access-management/v1/search/canonicalPrincipals/byResource?resource_type={{resourceType}}${permissionFilters && permissionFilters.length > 0 ? '&permissionFilter={{commaSeparatedPermissionFilter}}' : ''}`, {
+            resourceType,
+            commaSeparatedPermissionFilter: permissionFilters && permissionFilters.length > 0 && permissionFilters.join(','),
```

Comment:
Do you think there is a need to URL encode commaSeparatedPermissionFilter, or do you think that the grammar for permissions names will never be extended?

Could __buildUrl URL automatically encode all parameters?

Reply:
Discussed offline. We'll be doing this outside this PR.